### PR TITLE
Extend the GitHub API with requests to get `added` and `modified` files of a PR

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* add the `pontos-github file-status` command
* This command (currently) can print what files have been `modified` and `added` within the given PR
* Example: 
```
% pontos-github FS greenbone/python-gvm 662
ℹ pontos-github => file_status
✓     PR 662 is existing.
ℹ     Modified:
      /Users/jloechte/greenbone/pontos/gvm/protocols/gmpv2110/__init__.py
      /Users/jloechte/greenbone/pontos/tests/protocols/gmpv2110/entities/test_users.py
      /Users/jloechte/greenbone/pontos/tests/protocols/gmpv2110/entities/users/__init__.py
      /Users/jloechte/greenbone/pontos/tests/protocols/gmpv2110/entities/users/test_modify_user.py
ℹ     Added:
      /Users/jloechte/greenbone/pontos/gvm/protocols/gmpv2110/entities/users.py
      /Users/jloechte/greenbone/pontos/tests/protocols/gmpv2110/entities/users/test_create_user.py
```
* This can also be printed into a given output file
* Extended the API for the calls `pull_request_existing`, `pull_request_files`

**Why**:

**How**:

**Checklist**:

- [x] Tests
- [x] Conventional commit message
- [x] Documentation

### CODE REVIEW

File: pontos/github/api.py [hunk 1]
```diff
@@ -362,14 +374,14 @@ def pull_request_files(
         # possible to receive 100
         # might add the page parameter, to get the files 101-202 and so on
         params = {"per_page": "100"}
-        api = f"/repos/{repo}/pulls/{pr_number}/files"
+        api = f"/repos/{repo}/pulls/{pull_request}/files"
         response = self._request(api, params=params)
         file_dict = {}
         for status in status_list:
-            file_dict[status] = [
+            file_dict[status.value] = [
```
Comment (bjoernricks) [inline pontos/github/api.py:381]: I am really sure we want to use the `FileStatus` here
```suggestion
            file_dict[status] = [
```

File: pontos/github/api.py [hunk 2]
```diff
@@ -342,7 +354,7 @@ def download_release_zip(
         return download(api, destination)
 
     def pull_request_files(
-        self, repo: str, pr_number: int, status_list: List[str]
+        self, repo: str, pull_request: int, status_list: List[FileStatus]
     ) -> List[Path]:
```
Comment (bjoernricks) [inline pontos/github/api.py:358]:
```suggestion
    ) -> Dict[FileStatus, Iterable[Path]]:
```

File: pontos/github/cmds.py [hunk 3]
```diff
@@ -65,21 +65,23 @@ def file_status(args: Namespace):
...
-            repo=args.repo, pr_number=args.pr_number, status_list=args.status
+            repo=args.repo,
+            pull_request=args.pull_request,
+            status_list=args.status,
         )
         for status in args.status:
-            info(f'{status}:')
-            files = [str(f.resolve()) for f in file_dict[status]]
+            info(f'{status.value}:')
+            files = [str(f.resolve()) for f in file_dict[status.value]]
```
Comment (bjoernricks) [inline pontos/github/cmds.py:84]:
```suggestion
            files = [str(f.resolve()) for f in file_dict[status]]
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: tests/github/test_api.py [hunk 4]
```diff
@@ -329,6 +336,78 @@ def test_download_release_zip(
...
+                'Accept': 'application/vnd.github.v3+json',
+            },
+            params={'per_page': '100'},
+            json=None,
+        )
+
+        self.assertEqual(
+            files,
+            {
+                'modified': [
```
Comment (y0urself) [inline tests/github/test_api.py:364]:
```suggestion
                FileStatus.MODIFIED: [
```

File: tests/github/test_api.py [hunk 5]
```diff
@@ -329,6 +336,78 @@ def test_download_release_zip(
...
+                'Accept': 'application/vnd.github.v3+json',
+            },
+            params={'per_page': '100'},
+            json=None,
+        )
+
+        self.assertEqual(
+            files,
+            {
+                'added': [
```
Comment (y0urself) [inline tests/github/test_api.py:401]:
```suggestion
                FileStatus.ADDED: [
```

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**